### PR TITLE
docs(*): add disk usage docs

### LINKED ISSUE #2559: docs: Provide details about Deis disk usage and locations
@piotrb pointed out that it would be helpful to indicate where on the filesystem Deis stores data, and for what uses. This enables users to, for example, attach multiple volumes to their host optimized for different purposes (i.e. EC2 ephemeral disk for fast access, EBS for large but slow storage).

### PR DESCRIPTION
**Pull request by carmstrong**
closes #2559 
refs #1678 
refs #2886

### CODE REVIEW

**Comment by gabrtv** [comment]
Docs LGTM.  This is a good addition to the Managing Deis section.

**Comment by mboersma** [comment]
This will be a big help in troubleshooting disk space issues. LGTM.

File: docs/managing_deis/disk_usage.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+local storage on a machine.
+
+The following filesystem paths are the most important to consider:
+
+===================      =============================        ============================================================================================================================================
+location                 purpose                              considerations
+===================      =============================        ============================================================================================================================================
+/var/lib/etcd            etcd snapshot data                   etcd writes a relatively small amount of snapshot data here, so access should be as fast as possible (cloud providers use fast, local disks)
+/var/lib/docker          Docker image/volume storage          should be large - on cloud providers with external storage (AWS, GCE, Azure) this is a separate 100GB volume
+/var/lib/deis/store      mounted CephFS for deis-store        none - this is a virtually-mounted filesystem, and the "real" Ceph data lives in a Docker volume (so it's stored in /var/lib/docker)
```
Comment (bacongobbler) [inline docs/managing_deis/disk_usage.rst]: Don't the logs get shipped here (or anything using cephFS for that matter)? Some consideration should be presented here, even if only 5GB or so.

File: docs/managing_deis/disk_usage.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+The most alarming low-disk-space condition is when the Docker volume is nearly full. The ``builder``
+component should remove unnecessary images after a build, and will also remove images after
+an application has been deleted.
+
+However, it some cases it may be necessary to manually prune old images using ``docker rmi``:
+
+.. code-block:: console
+
+    $ docker images -aq | xargs -l10 docker rmi
```
Comment (bacongobbler) [inline docs/managing_deis/disk_usage.rst]: A `deisctl restart builder` will also purge the builder of all images persisting within its walls, which should further clean up disk space.

File: docs/managing_deis/disk_usage.rst (hunk 1 above)
Comment (carmstrong) [inline docs/managing_deis/disk_usage.rst]: It's not a real filesystem - users don't manage this at all. That's exactly what the "consideration" comment is for.

File: docs/managing_deis/disk_usage.rst (hunk 2 above)
Comment (carmstrong) [inline docs/managing_deis/disk_usage.rst]: Good to note, but not sure I want to put it in the docs - seems like a sort of clunky workaround as resetting every app's git cache isn't always desired.

**Pull request merged**